egui_plot/src/items/mod.rs

```diff
@@ -518,11 +566,24 @@ impl PlotItem for Line<'_> {
             final_stroke = PathStroke::new_uv(stroke.width, wrapped_callback.clone());
         }
 
-        let values_tf: Vec<_> = series
-            .points()
-            .iter()
-            .map(|v| transform.position_from_point(v))
-            .collect();
+        let mut values_tf: Vec<Pos2> = Vec::new();
```

Comment:
This should be avoided - do not construct new vec from the series, that will make a new unnecessary allocation.